### DISCUSSION TITLE
Add Naming or Subject Line to Threads

### DISCUSSION THREAD

**Discussion by matthewbirtch** [mattermost/community#36] [Enhancements]
With Collapsed Reply Threads now in Beta, it would be nice to have the ability to optionally give a thread a topic title or subject line to help scan and identify the thread quickly

**Comment by aedott** [comment]
This would be really helpful! I sometimes use markdown headers to achieve the same effect, but it would be nice if it were slightly more formalized. It would also be great if there were a way to see a list of all thread subject lines in a channel, so you don't have to scroll back forever to find a thread from weeks ago.

**Comment by larkox** [comment]
This sounds cool. But would it be per user (I name my threads, and nobody see those names) or by thread (the thread creator names the thread, and everyone see the same name)? Or both?

**Reply by gigawhitlocks** [reply]
I believe the suggestion is for everyone to see the thread name. As for permissions regarding who can edit the subject, I think that would depend on the edition of Mattermost, right? E.g. on TE anyone could rename any thread, much as they can with any channel, but EE might have more granular permissions along with the EE permissions features that already exist

**Comment by esethna** [comment]
Permissions and discoverability are the hardest challenges with this idea but I think it's something we should absolutely do. I don't think we want threads to require a title, but giving users with the right permissions an option to name it from the RHS could potentially  work

**Comment by Willyfrog** [comment]
I agree with @larkox that naming should be a personal thing. Some people have their own way to organize and that might clash with other people's system.

**Comment by matthewbirtch** [comment]
I think it's important that all users see the same name for a thread - not just your own personal name. In the same way that channel names are the same for everyone, we would be introducing an inconsistent mental model if thread names were personal to you. There is also the previous convention with email and other tools where subject lines are used. I believe it would be quite unexpected for the majority of users if thread naming was personal.

Having said that, I think we might consider allowing the thread name/subject line to be modified by more than just the root post author. Something we'll need to investigate

**Reply by esethna** [reply]
+1 to Matt's comments and the similarities to email

**Reply by Willyfrog** [reply]
I'm kind of bought on your line of thought, but then... Can we have categories? That way we get a meaningful naming for everyone and a way to sort threads

**Comment by chetanyakan** [comment]
Can be implemented in 2 ways:
- A personal note on the thread topic (visible to yourself but not others).
- The topic visible to all users.

**Reply by esethna** [reply]
Agree, there could be two approaches to this, although I don't know that we want to offer both (I could see that getting confusing)
